src/Acorn/View/Composer.php

```diff
@@ -32,26 +48,27 @@ public static function views()
     /**
      * Compose the view before rendering.
      *
-     * @param \Illuminate\View\View $view;
+     * @param  \Illuminate\View\View $view
      * @return void
      */
     public function compose(View $view)
     {
         $this->view = $view;
+        $this->data = new Fluent($view->getData());
 
-        $view->with($this->getData());
+        $view->with($this->merge());
     }
 
     /**
-     * Data to be passed to view before rendering
+     * Data to be merged and passed to the view before rendering.
      *
      * @return array
      */
-    protected function getData()
+    protected function merge()
     {
         return array_merge(
             $this->with(),
-            $this->view->getData(),
+            $this->data->toArray(),
```

Comment:
I think we should leave this as `$this->view->getData()`. The idea is that the view data remains untouched except what is being returned by `with()` and `override()` methods. By using `$this->data` here, it means that modifying `$this->data` within a view composer, you're directly modifying the data that's getting passed back to the view. It defeats the purpose of having separate `with()` and `override()` methods. Maybe that's desirable, I don't know; we should discuss it. It should be a separate issue/PR imo.